bundles/org.openhab.binding.opensmartcity/src/main/java/org/openhab/binding/opensmartcity/internal/OpenSmartCityWeatherHandler.java

```diff
@@ -74,19 +88,34 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void initialize() {
+        boolean validConfig = true;
         config = getConfigAs(OpenSmartCityWeatherConfiguration.class);
-
-        bridgeHandler = (OpenSmartCityCityHandler) getBridge().getHandler();
-
-        ScheduledFuture<?> localRefreshJob = refreshJob;
-        if (localRefreshJob == null || localRefreshJob.isCancelled()) {
-            logger.debug("Start refresh job at interval {} seconds.", bridgeHandler.config.refreshInterval);
-            refreshJob = scheduler.scheduleWithFixedDelay(this::updateSensorValues, 0,
-                    bridgeHandler.config.refreshInterval, TimeUnit.SECONDS);
+        String thingUid = getThing().getUID().toString();
+        myLoc = config.location;
+        if (myLoc.isEmpty()) {
+            logger.debug("Geo Location is mandatory and has to be set in system settings, disabling thing {}",
+                    thingUid);
+            validConfig = false;
+        }
+        if (validConfig) {
+            updateStatus(ThingStatus.ONLINE);
+            bridgeHandler = (OpenSmartCityCityHandler) getBridge().getHandler();
+
+            ScheduledFuture<?> localRefreshJob = refreshJob;
+            if (localRefreshJob == null || localRefreshJob.isCancelled()) {
+                logger.debug("Start refresh job at interval {} seconds.", bridgeHandler.config.refreshInterval);
+                refreshJob = scheduler.scheduleWithFixedDelay(this::updateSensorValues, 0,
+                        bridgeHandler.config.refreshInterval, TimeUnit.SECONDS);
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
         }
     }
 
     private void updateSensorValues() {
+
+        getNearestOnlineLocation();
```

Comment:
Looks expensive to do this on every single refresh. Isn't it enough to get the closest station at initialization and then only update the list in case it goes offline?

Reply:
No, what happens if they take the Thing offline and we don't get new values for a longer period?
I deeply thought about that, but there is no other option.

Reply:
How about only checking the online status of the sensor (via api call) at start of the update method and only get the nearest location in case it isn't available anymore?

As you said "they" will put it offline or in maintenance, so we will definetely gat some sort of informatoin from the api in any case.

Reply:
Did not think about it that way.
Seems to be feasable.
@Confectrician Can you create the call to check if a given Thing is Online? Don't think we can go down to sensor level....

Reply:
@kaikreuzer @Confectrician 
Implemented 😉